sgbatch/src/computation/CacheComputation.cpp

```diff
@@ -90,81 +93,34 @@ void StreamedComputation::determineFileSources(std::string hostname) {
-    while ((SimpleSimulator::mean_flops_per_block + SimpleSimulator::sigma_flops_per_block) < dflops || dflops < 0.)
-        dflops = (*SimpleSimulator::flops_per_block_dist)(SimpleSimulator::gen);
-    return dflops * num_blocks;
 }
 
-void StreamedComputation::performComputationNoStreaming(std::string &hostname) {
-    WRENCH_INFO("Performing non-streamed computation!");
-    // Read all input files
-    double data_size = 0;
-    for (auto const &fs : this->file_sources) {
-        WRENCH_INFO("Reading file %s from storage service on host %s",
-                    fs.first->getID().c_str(), fs.second->getStorageService()->getHostname().c_str());
-        fs.second->getStorageService()->readFile(fs.first, fs.second);
-        data_size += fs.first->getSize();
-    }
-    // Perform the computation as needed
-    double flops = determineFlops(data_size);
-    WRENCH_INFO("Computing %.2lf flops", flops);
-    wrench::Simulation::compute(flops);
+double CacheComputation::determineFlops(double data_size, double total_data_size) {
+    double flops = this->total_flops * data_size / total_data_size;
+    return flops;
 }
 
-void StreamedComputation::performComputationStreaming(std::string &hostname) {
-    WRENCH_INFO("Performing streamed computation!");
-    for (auto const &fs : this->file_sources) {
-        WRENCH_INFO("Streaming computation for input file %s", fs.first->getID().c_str());
-        double data_to_process = fs.first->getSize();
-
-        // Compute the number of blocks
-        int num_blocks = int(std::ceil(data_to_process / (double) SimpleSimulator::xrd_block_size));
-
-        // Read the first block
-        fs.second->getStorageService()->readFile(fs.first, fs.second, std::min<double>(SimpleSimulator::xrd_block_size, data_to_process));
-
-        // Process next blocks: compute block i while reading block i+i
-        for (int i=0; i < num_blocks - 1; i++) {
-            double num_bytes = std::min<double>(SimpleSimulator::xrd_block_size, data_to_process);
-            double num_flops = determineFlops(num_bytes);
-//            WRENCH_INFO("Chunk: %.2lf bytes / %.2lf flops", num_bytes, num_flops);
-            // Start the computation asynchronously
-            simgrid::s4u::ExecPtr exec = simgrid::s4u::this_actor::exec_init(num_flops);
-            exec->start();
-            // Read data from the file
-            fs.second->getStorageService()->readFile(fs.first, fs.second, num_bytes);
-            // Wait for the computation to be done
-            exec->wait();
-            data_to_process -= num_bytes;
-        }
-
-        // Process last blocks
-        double num_flops = determineFlops(std::min<double>(SimpleSimulator::xrd_block_size, data_to_process));
-        simgrid::s4u::ExecPtr exec = simgrid::s4u::this_actor::exec_init(num_flops);
-        exec->start();
-        exec->wait();
-
-    }
-
+void CacheComputation::performComputation(std::string &hostname) {
```

Comment:
@henricasanova is this the right way to do this?

Reply:
sure. that's ok. The 3 lines could be written instead as: wrench::Simulation::compute(num_flops); 
 (We do the simgrid stuff above because we need asynchronous-ness, which WRENCH doesn't expose right now, but if it's just a synchronous computation, might as well use the Simulation::compute(), much simpler, method